internal/hooks/hook_executor_default.go

```diff
@@ -31,7 +31,7 @@ type HookExecutorDefaultProtocol struct {
 }
 
 // Execute processes the data received by the SDK.
-func (e *HookExecutorDefaultProtocol) Execute(opts HookExecOpts) (string, error) {
+func (e *HookExecutorDefaultProtocol) Execute(ctx context.Context, opts HookExecOpts) (string, error) {
```

Comment:
⭐ ⭐ ⭐ 
note: This is 1 of 2 key changes in this PR. Here we update the default hook executor protocol to accept a context argument. This allows us to remove the `context.Background()` usage inside this function.
⭐ ⭐ ⭐

Reply:
@mwbrooks Fantastic catch here!

IIRC a "background" context was used as a quick workaround for debug printing. It's now more clear that a follow up for this is important and having it broken into a standalone PR makes reviewing swell 🫡